### DISCUSSION TITLE
Add a possibility to catch webrtc ice connection process errors and more

### DISCUSSION THREAD

**Discussion by tlloancy** [godotengine/godot-proposals#9599] [Engine Core]
### Describe the project you are working on

A P2P contact system where the user is able to select his own ice server

### Describe the problem or limitation you are having in your project

i can't catch any error ice related, as described here https://github.com/godotengine/webrtc-native/issues/145

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If i could catch those errors, i could display them on ui and it would greatly help for both UI and DEVELOPMENT purposes

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

its just a try{} catch(e) {} or except() system to not let the error crash the system and be able to display it instead

### If this enhancement will not be used often, can it be worked around with a few lines of script?

not sure about that at all

### Is there a reason why this should be core and not an add-on in the asset library?

i dont knwo about that thats why i have opened both issues https://github.com/godotengine/webrtc-native/issues/145 https://github.com/godotengine/godot/issues/84645

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Comment by tlloancy** [comment]
Hi, by all means im probably out of your league on those c++ libraries + gdscript combo. But i tried a simple python example which show what i'd expect (ouioui(0)) and what i actually get (ouioui(1)). And i think this is because you already catch errors in webrtc, but there is no cascada.  (ouioui(2))

~~https://www.online-python.com/JNilDo7PIc~~
https://www.online-python.com/PpKo3xUMcn

**Comment by tlloancy** [comment]
just a quick explanation here

call to functions with main.py

```

# Online Python - IDE, Editor, Compiler, Interpreter
from Untitled3 import *


try:
    ouioui(0)
except Exception as e:
    print("error: " + repr(e))
    
try:
    ouioui(1)
except Exception as e:
    print("Id like to catch you, but i just can't" + repr(e))
    
try:
    ouioui(2)
except Exception as e:
    print("HERE " + repr(e))
```

where is emulated the call to webrtc lib is Untitled3.py
```
from Untitled2 import *
def ouioui(a):
    if a == 0:
        raise Exception("ice url invalid")
    else:
        try:
            olol()
        except Exception as e:
            print(repr(e))
            if a == 2:
                raise Exception(repr(e))
    #olol()["toto"] = 4
```

Error can be catched if reraised as shown in Untitled2 my crew

```
def olol():
  raise Exception("i fail but im catched")  
#    return None
```

And we're done

**Comment by tlloancy** [comment]
Usually, THAT is the output we get:

try 1:
```
error: Exception('ice url invalid')
```

try 2:
```
Exception('i fail but im catched')
```
(not really indeed)

try 3:
```
Exception('i fail but im catched')
HERE Exception("Exception('i fail but im catched')")
```

There we go !

**Comment by tlloancy** [comment]
Hi so is there a way to do that that i ignore or does this need an update

**Comment by tlloancy** [comment]
allo ?
